shaptools/hdb_connector/connectors/pyhdb_connector.py

```diff
@@ -49,27 +49,26 @@ def connect(self, host, port=30015, **kwargs):
             )
         except socket.error as err:
             raise base_connector.ConnectionError('connection failed: {}'.format(err))
-        self._logger.info('connected succesfully')
+        self._logger.info('connected successfully')
 
     def query(self, sql_statement):
         """
-        Query a sql statement and return response
+        Query a sql query result and return  a result object
         """
         self._logger.info('executing sql query: %s' % sql_statement)
         try:
             cursor = self._connection.cursor()
             cursor.execute(sql_statement)
-            result = cursor.fetchall()
+            result = result = base_connector.QueryResult().fetch_query(cursor)
             cursor.close()
         except pyhdb.exceptions.DatabaseError as err:
             raise base_connector.QueryError('query failed: {}'.format(err))
-        self._logger.info('query result: %s' % result)
         return result
```

Comment:
Here we have to add the `finally` keyword (same indentation level as `try` and `except`) and close the cursor there